File: 3172-divisible-and-non-divisible-sums-difference/solution.java
class Solution {
    public int differenceOfSums(int n, int m) {
        int s=0,k=0;
        for(int i=1;i<=n;i++)
        {
            if(i%m==0)
            {
                s+=i;
            }
            else
            {
                k+=i;
            }
        }
        return k-s;
        
    }
}

File: 3172-divisible-and-non-divisible-sums-difference/solution.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        k = sum(i for i in range(1,n+1) if i%m!=0)
        y = sum(i for i in range(1,n+1) if i%m==0)
        return k-y

